src/cloudWatchLogs/utils.ts

```diff
@@ -0,0 +1,27 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { CloudWatchLogs } from 'aws-sdk'
+import * as vscode from 'vscode'
+import * as nls from 'vscode-nls'
+const localize = nls.loadMessageBundle()
+
+import { CloudWatchLogsClient } from '../shared/clients/cloudWatchLogsClient'
+
+export async function* listCloudWatchLogGroups(
+    client: CloudWatchLogsClient
+): AsyncIterableIterator<CloudWatchLogs.LogGroup> {
+    const status = vscode.window.setStatusBarMessage(
```

Comment:
I'd prefer to avoid copying this pattern for new code (and address it in existing code separately)...

The problem is if a caller stops iterating before the full result set is returned, the finally block is never entered, meaning the status bar doesn't go away. The status bar would be a concern of the caller.

Reply:
Interesting, any proposed ways around that? I'm assuming we don't want to drop the status bar?

Reply:
We might be able to start without the status bars. The Tree nodes already have a loading indicator while they are getting data.